pennylane/spin/spin_hamiltonian.py

```diff
@@ -304,7 +304,307 @@ def fermi_hubbard(
+            {(0, up_spin): "+", (1, up_spin): "-", (2, down_spin): "+", (3, down_spin): "-"}
+        )
+
+    hamiltonian = hopping_ham + coulomb_term + intersite_term
+
+    if mapping not in ["jordan_wigner", "parity", "bravyi_kitaev"]:
+        raise ValueError(
+            f"The '{mapping}' transformation is not available."
+            f"Please set mapping to 'jordan_wigner', 'parity', or 'bravyi_kitaev'."
+        )
+    qubit_ham = qml.qchem.qubit_observable(hamiltonian, mapping=mapping)
+
+    return qubit_ham.simplify()
+
+
+def haldane(
+    lattice,
+    n_cells,
+    hopping1=1.0,
+    hopping2=1.0,
+    phi=1.0,
+    boundary_condition=False,
+    mapping="jordan_wigner",
+):
+    r"""Generates the `Haldane model <https://arxiv.org/pdf/2211.13615>`_ Hamiltonian on a lattice.
+
+    The Hamiltonian is represented as:
+
+    .. math::
+
+        \begin{align*}
+          \hat{H} & = -\sum_{\langle i,j \rangle}t_{ij}^{1}
+          (c_{i\sigma}^\dagger c_{j\sigma} + c_{j\sigma}^\dagger c_{i\sigma})
+          - \sum_{\langle\langle i,j \rangle\rangle, \sigma} t_{ij}^{2}
+          \left( e^{i\phi_{ij}} c_{i\sigma}^\dagger c_{j\sigma} + e^{-i\phi_{ij}} c_{j\sigma}^\dagger c_{i\sigma} \right)
+        \end{align*}
+
+    where :math:`t^{1}_{ij}` is the hopping term representing the hopping amplitude between neighbouring
+    sites :math:`\langle i,j \rangle`, :math:`t^{2}_{ij}` is the hopping amplitude between next-nearest neighbours :math:`\langle \langle i,j \rangle \rangle`, :math:`\phi_{ij}` is the phase
+    factor that breaks time-reversal symmetry in the system, and :math:`\sigma` is the spin degree of freedom.
+    This function assumes there are two fermions with opposite spins on each lattice site.
+
+    Args:
+        lattice (str): Shape of the lattice. Input values can be ``'chain'``, ``'square'``,
+            ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, or ``'kagome'``.
+        n_cells (list[int]): Number of cells in each direction of the grid.
+        hopping1 (Optional[float | tensor_like(float)]): Hopping strength between
+            nearest neighbouring sites. It can be a number, or
+            a square matrix of size ``(n_sites, n_sites)``, where ``n_sites`` is the total
+            number of sites. Default value is 1.0.
+        hopping2 (Optional[float | tensor_like(float)]): Hopping strength between next-nearest
+            neighbouring sites. It can be a number, or
+            a square matrix of size ``(n_sites, n_sites)``, where ``n_sites`` is the total
+            number of sites. Default value is 1.0.
+        phi (Optional[float | tensor_like(float)]): Phase factor in the system. It can be a number, or
+            a square matrix of size ``(n_sites, n_sites)``, where ``n_sites`` is the total
+            number of sites. Default value is 1.0.
+        boundary_condition (Optional[bool | list[bool]]): Defines boundary conditions for different lattice
```

Comment:
```suggestion
        hopping1 (float | tensor_like(float)): Hopping strength between
            nearest neighbouring sites. It can be a number, or
            a square matrix of size ``(n_sites, n_sites)``, where ``n_sites`` is the total
            number of sites. Default value is 1.0.
        hopping2 (float | tensor_like(float)): Hopping strength between next-nearest
            neighbouring sites. It can be a number, or
            a square matrix of size ``(n_sites, n_sites)``, where ``n_sites`` is the total
            number of sites. Default value is 1.0.
        phi (float | tensor_like(float)): Phase factor in the system. It can be a number, or
            a square matrix of size ``(n_sites, n_sites)``, where ``n_sites`` is the total
            number of sites. Default value is 1.0.
        boundary_condition (bool | list[bool]): Defines boundary conditions for different lattice
```